chiron/potential.py

```diff
@@ -0,0 +1,96 @@
+import jax
+import jax.numpy as jnp
+from scipy.spatial.distance import pdist, squareform
+from loguru import logger as log
+from openmm import unit
+from openmm.app import Topology
+import jax.numpy as jnp
+
+
+class NeuralNetworkPotential:
+    def __init__(self, model, topology: Topology):
+        self.topology = topology  # The topology of the system
+        if model is None:
+            log.warning("No model provided, using default model")
+        else:
+            self.model = model(self.topology)  # The trained neural network model
+
+    def compute_energy(self, positions):
+        # Compute the pair distances and displacement vectors
+        pair_distances = pdist(positions)
+        displacement_vectors = squareform(pair_distances)
+        # Use the model to compute the potential energy
+        potential_energy = self.model(displacement_vectors)
+        return potential_energy
+
+    def compute_force(self, positions) -> jnp.ndarray:
+        # Compute the force as the negative gradient of the potential energy
+        positions_without_unit = jnp.array(
+            positions.value_in_unit_system(unit.md_unit_system)
+        )
+        force = -jax.grad(self.compute_energy)(positions_without_unit)
+        return force
+
+    def compute_pairlist(self, positions, cutoff) -> jnp.array:
```

Comment:
I know this is still a draft PR, but I don't think this bit of code is going to actually provide the list of pairs.  While this will give us a list of which pairs have distances within the cutoff, it doesn't actually return the pairs themselves. 

We need to precompute the possible pairs in the same order that scipy will use, which is basically just a nested loop
```
pairs = jnp.array( [ (i,j) for I in range(0, positions.shape[0]) for j in range(i+1, positions.shape[0])])

pair_distances = pdist(positions)
pairlist = pairs[ jnp.where(pair_distances< cutoff)]

return pairlist
```

or we could use cdist which gives us the matrix of all possible pairs
```
pair_distances = cdist(positions, positions)
pairs = jnp.where(pair_distances < cutoff)

# exclude case where i ==j and duplicate pairs  
mask = jnp.where(pairs[0] < pairs[1])

return pairs[0][mask], pairs[1][mask]
```

Reply:
That's great, thank you!
I was aware of that, but I wanted to focus on the HO first anywas, that was the reason why I just left it there.
But thanks a lot for the suggestion, I will update the code here.